lib/doctest.js

```diff
@@ -523,50 +562,113 @@ function commonjsEval(source, path) {
   } finally {
     fs.unlinkSync (abspath);
   }
-  return run (queue);
+  return run (options, queue, options);
 }
 
-function run(queue) {
-  return queue.reduce (function(accum, io) {
-    var thunk = accum.thunk;
-    if (io.type === INPUT) {
-      if (thunk != null) thunk ();
-      accum.thunk = io.thunk;
-    } else if (io.type === OUTPUT) {
-      var either;
+function run(options, queue, optionalLogMediator) {
+  var logMediator = optionalLogMediator || {};
```

Comment:
Could this be moved to the call sites? In other words, could we *require* a log mediator?

Reply:
This might be a leftover from an earlier version where I didn't yet pass `options` down, and the absence of the logMediator would indicate that logging is not enabled.

Reply:
Ah, no. It's because some logic hasn't yet been updated to supply a logMediator. This code should go away once all source types implement logging.